content/docs/user-guide/troubleshooting.md

```diff
@@ -125,3 +125,39 @@ and retry the DVC command. Specifically, one of:
 
 [internal directories]:
   https://dvc.org/doc/user-guide/project-structure/internal-files
+
+## DVC experiments may fail in shallow Git repositories {#git-shallow}
+
+DVC experiments rely on Git features which may not work properly in shallow Git
+repositories. This error commonly occurs in certain CI environments which use
+shallow git clones by default. When this error is encountered, try unshallowing
+the Git repository and then retry the DVC command.
+
+Local Git repositories can be unshallowed by using:
+
+```
+$ git fetch --unshallow
+```
```

Comment:
```suggestion
`dvc exp` commands use internal Git operations which may not work properly in
[shallow clones](https://git-scm.com/docs/git-clone#Documentation/git-clone.txt---depthltdepthgt).
Local Git repositories can be unshallowed with `git fetch --unshallow`.

Typically, however, this occurs in transient remote environments such as
Continuous Integration jobs, which use shallow clones by default. In those cases,
change their configuration to avoid shallow cloning. Common examples:
```